src/arrays.js

```diff
@@ -1,61 +1,133 @@
+  for (let i = 0; i < numbers.length; i += 1){
+    strings.push(numbers[i].toString());
+  }
+  return strings;
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  let uppercaseArray = [];
+  for (let i = 0; i < strings.length; i += 1){
+    uppercaseArray.push(strings[i].toUpperCase());
+  }
+  return uppercaseArray;
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+  /*A very difficult problem to solve for me, after some googling, I found the following:
+  https://stackoverflow.com/questions/59621299/javascript-reversing-every-string-of-an-array
+  Solution below is a simplified version of what the person asking the question got to in the end */
+  let reversedArray = [];
+  //First for loop loops through the array and processes each string
+  for (let i = 0; i < strings.length; i += 1){
+    /*Second for loop stores the letters in reverse, starting with the last character and moving down.
+    j has to be 1 less than the word's length and decreases by 1 each time, and the loop stops it reaches character 0 */
+    let reversedWord = '';
+    for (let j = strings[i].length - 1; j >= 0; j -= 1){
+      reversedWord += strings[i][j];
+    }
+    reversedArray.push(reversedWord);
+  }
+  return reversedArray;
 };
 
 const onlyEven = numbers => {
-  // your code here
+  let evenNumbers = [];
+  for (let i = 0; i < numbers.length; i += 1){
+    /*An even number is any number that gives no remainder when divided by two
+    Only numbers meeting that condition are pushed into the new array */
+    if (numbers[i] % 2 === 0){
+      evenNumbers.push(numbers[i]);
+    }
+  }
+  return evenNumbers;
 };
 
 const removeNthElement2 = (index, array) => {
-  // your code here
+  let shorterArray = [...array];
+  shorterArray.splice(index, 1);
+  return shorterArray;
 };
 
 const elementsStartingWithAVowel = strings => {
-  // your code here
+  /*https://stackoverflow.com/questions/52028403/filter-array-of-strings-keeping-only-ones-starting-with-vowels
+  /^[aeiou]/i is basically asking to match a/e/i/o/u at the start and the /i part makes it case-insensitive
+  Those that don't begin with a vowel are .filter -ed out
+  At this stage, still not 100% sure how this works entirely */
+  return strings.filter(str => /^[aeiou]/i.test(str));
 };
 
 const removeSpaces = string => {
-  // your code here
+  let spacelessString = '';
+  /*For loop rolls through every character in the string, checking that it's NOT an empty space
+  Because it rolls through every character, sneaky spaces have nowhere to hide here
+  If the character is NOT an empty space, it gets added to the spacelessString variable */
+  for (i = 0; i < string.length; i += 1){
+    if(string[i] !== ' '){
+      spacelessString += string[i]
+    }
+  }
+  return spacelessString;
 };
 
 const sumNumbers = numbers => {
-  // your code here
+  let sum = 0;
+  //For loop rolls through array and adds each element to sum until all elements are added
```

Comment:
This is a great solution with a for loop and it works. Try refactoring it using one of the iterator methods we learned on Wednesday (hint: reduce!)